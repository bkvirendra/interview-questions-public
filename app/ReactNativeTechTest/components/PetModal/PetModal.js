import React from "react";
import { Modal, StyleSheet, Text, Pressable, View } from "react-native";
import pluralize from "pluralize";

const PetModal = ({isVisible, onHide, pet, onDelete}) => {
  if (!isVisible) return null;
  return (
    <View style={styles.centeredView}>
      <Modal
        animationType="slide"
        transparent={true}
        visible={isVisible}
      >
        <View style={styles.centeredView}>
          <View style={styles.modalView}>
            <View style={styles.iconContainer} />
            <Text style={[styles.modalText, styles.modalTitleText]}>{pet.name}</Text>
            <Text style={styles.modalText}>{pet.type}</Text>
            <Text style={styles.modalText}>Needs feeding {pluralize('times', pet.feeds, true)} a day!</Text>
            <Pressable
              style={[styles.button, styles.buttonClose]}
              onPress={onHide}
            >
              <Text style={styles.textStyle}>Close</Text>
            </Pressable>
            <Pressable
              style={[styles.button, styles.buttonDelete]}
              onPress={onDelete}
            >
              <Text style={styles.textStyle}>Delete</Text>
            </Pressable>
          </View>
        </View>
      </Modal>
    </View>
  );
};

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22
  },
  iconContainer: {
    marginTop: -80,
    marginBottom: 24,
    height: 90,
    width: 90,
    borderRadius: 45,
    backgroundColor: 'grey',
  },
  modalView: {
    margin: 20,
    backgroundColor: "white",
    borderRadius: 20,
    padding: 35,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5
  },
  button: {
    borderRadius: 20,
    padding: 10,
    elevation: 2,
    margin: 4,
  },
  buttonDelete: {
    backgroundColor: "#96281b",
  },
  buttonClose: {
    backgroundColor: "#2196F3",
  },
  textStyle: {
    color: "white",
    fontWeight: "bold",
    textAlign: "center"
  },
  modalText: {
    marginBottom: 15,
    textAlign: "center"
  },
  modalTitleText: {
    fontWeight: "bold"
  }
});


export default PetModal;
