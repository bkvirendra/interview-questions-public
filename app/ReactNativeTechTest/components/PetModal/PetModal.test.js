/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import PetModal from './PetModal';

describe('Components - PetModal', () => {
  it('renders - light mode', () => {
    const petObject = {
      name: 'Bowser',
      type: 'Rottweiler',
      feeds: 4,
      id: 1,
    };
    const tree = renderer
      .create(<PetModal isVisible={true} pet={petObject} />)
      .toJSON();
    expect(tree).toMatchSnapshot();

    const modalHiddenTree = renderer
      .create(<PetModal isVisible={false} pet={petObject} />)
      .toJSON();
    expect(modalHiddenTree).toMatchInlineSnapshot(`null`);
  });
});
