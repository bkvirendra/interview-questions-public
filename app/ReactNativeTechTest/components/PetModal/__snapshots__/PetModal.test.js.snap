// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - PetModal renders - light mode 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 22,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "elevation": 5,
            "margin": 20,
            "padding": 35,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "grey",
              "borderRadius": 45,
              "height": 90,
              "marginBottom": 24,
              "marginTop": -80,
              "width": 90,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "marginBottom": 15,
                "textAlign": "center",
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          Bowser
        </Text>
        <Text
          style={
            Object {
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
        >
          Rottweiler
        </Text>
        <Text
          style={
            Object {
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
        >
          Needs feeding 
          4 times
           a day!
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 20,
                "elevation": 2,
                "margin": 4,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#2196F3",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Close
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 20,
                "elevation": 2,
                "margin": 4,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#96281b",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Delete
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
